### PR TITLE
Init IMU before getting value

### PR DESCRIPTION
**Pull request by DanielHo-BS**
Update Main.c 
  Fix the spi_transfer issue about the "len".
  Add delay_time  in spi_writer.
  Show more details about self_test.
  Add example: 
    1. Get CHip ID.
    2. Get PUM status.
    3. Read  ACC and GYR data.

Add the file "main"

### CODE REVIEW

File: main.c [hunk 1]
```diff
@@ -103,12 +106,67 @@ int main(int argc, char argv[]) {
...
+	{
+		printf("Test Faild. Check the Fail Value:%d\n",rslt);
+	}
+	else {
+		pabort("Self Test Error!!");
 	}
+
+
+	/* Set the Accel power mode as normal mode */
+    sensor.accel_cfg.power = BMI160_ACCEL_NORMAL_MODE;
```
Comment (evshary) [inline main.c]: Should use the same indent.

File: main.c [hunk 2]
```diff
@@ -103,12 +106,67 @@ int main(int argc, char argv[]) {
 	int8_t rslt = BMI160_OK;
 	rslt = bmi160_init(&sensor);
 	if (rslt != BMI160_OK) pabort("Init Module Failed");
+	
+	/*!
+	After self test, all register values are overwritten with "default parameters".
+	*/
 	rslt = bmi160_perform_self_test(BMI160_ACCEL_ONLY, &sensor);
-	if (rslt >= BMI160_OK) {
+	if (rslt == BMI160_OK) {
```
Comment (evshary) [inline main.c:114]: Should use the same coding style.

**Review by evshary — CHANGES_REQUESTED** [review]

**Comment by evshary** [comment]
Besides, the binary main should not be put on repository.
Please use `.gitignore` to ignore it.

File: main.c (hunk 2 above)
Comment (DanielHo-BS) [reply main.c:114]: OK

File: main.c (hunk 1 above)
Comment (DanielHo-BS) [reply main.c]: Fixed

**Review by DanielHo-BS — COMMENTED** [review]
Done the require changes.

**Comment by evshary** [comment]
LGTM

**Pull request merged**